### PR TITLE
minimal recovery message from f to f+1

### PR DESCRIPTION
**Pull request by Liaojinghui**
#722

### CODE REVIEW

**Comment by Liaojinghui** [comment]
@fyrchik

**Review by fyrchik — APPROVED** [review]

**Review by shargon — APPROVED** [review]

**Review by vncoelho — CHANGES_REQUESTED** [review]
I will double check.

**Comment by vncoelho** [comment]
Give me 1 day just to double check and verify index and history.

**Review by vncoelho — APPROVED** [review]
In fact, this logic has passed through some modifications since dBFT 2.0 and this case reported has been missed.

File: src/DBFTPlugin/Consensus/ConsensusService.OnMessage.cs [hunk 1]
```diff
@@ -287,7 +287,7 @@ private void OnRecoveryRequestReceived(ExtensiblePayload payload, ConsensusMessa
             if (!context.CommitSent)
             {
                 bool shouldSendRecovery = false;
-                int allowedRecoveryNodeCount = context.F;
+                int allowedRecoveryNodeCount = context.F + 1;
                 // Limit recoveries to be sent from an upper limit of `f` nodes
```
Comment (vncoelho) [inline src/DBFTPlugin/Consensus/ConsensusService.OnMessage.cs:291]:
```suggestion
                // Limit recoveries to be sent from an upper limit of `f + 1` nodes
```

**Review by vncoelho — APPROVED** [review]

**Pull request merged**